auth/services/oauth/oauth.go

```diff
@@ -112,21 +112,26 @@ func (s *service) CreateAccessToken(request services.CreateAccessTokenRequest) (
 
 	// Validate the AuthTokenContainer, according to RFC003 §5.2.1.5
 	var err error
+
 	if context.jwtBearerTokenClaims.UserIdentity != nil {
 		var decoded []byte
+
 		if decoded, err = base64.StdEncoding.DecodeString(*context.jwtBearerTokenClaims.UserIdentity); err != nil {
 			return nil, fmt.Errorf("failed to decode base64 usi field: %w", err)
 		}
+
 		if context.contractVerificationResult, err = s.contractClient.VerifyVP(decoded, nil); err != nil {
 			return nil, fmt.Errorf("identity verification failed: %w", err)
 		}
-	}
-	if context.contractVerificationResult.Validity == contract.Invalid {
-		return nil, errors.New("identity validation failed")
-	}
-	// checks if the name from the login contract matches with the registered name of the issuer.
-	if err := s.validateActor(&context); err != nil {
-		return nil, err
+
+		if context.contractVerificationResult.Validity == contract.Invalid {
```

Comment:
I know this is existing code, but wouldn't it be safer to reverse this check, to fail if `Validity != contract.Valid`? Then it's also deemed invalid when a bug (or attack) occurs (e.g. `""`)

Reply:
I was actually thinking about something like this as well. You're right, will fix